hadoop-tools/hadoop-azure/src/test/java/org/apache/hadoop/fs/azurebfs/ITestGetNameSpaceEnabled.java

```diff
@@ -217,4 +226,31 @@ private AbfsClient callAbfsGetIsNamespaceEnabledAndReturnMockAbfsClient()
     return mockClient;
   }
 
+  @Test
+  public void ensureGetAclDetermineHnsStatusAccurately() throws Exception {
+    ensureGetAclDetermineHnsStatusAccuratelyInternal(HTTP_BAD_REQUEST, false);
+    ensureGetAclDetermineHnsStatusAccuratelyInternal(HTTP_NOT_FOUND, true);
+    ensureGetAclDetermineHnsStatusAccuratelyInternal(HTTP_INTERNAL_ERROR, false);
+    ensureGetAclDetermineHnsStatusAccuratelyInternal(HTTP_UNAVAILABLE, false);
+  }
+
+  private void ensureGetAclDetermineHnsStatusAccuratelyInternal(int statusCode,
+      boolean expected) throws Exception {
+    AzureBlobFileSystemStore store = Mockito.spy(getFileSystem().getAbfsStore());
+    AbfsClient mockClient = mock(AbfsClient.class);
+    store.setNamespaceEnabled(Trilean.UNKNOWN);
+    doReturn(mockClient).when(store).getClient();
+    AbfsRestOperationException ex = new AbfsRestOperationException(
+        statusCode, null, Integer.toString(statusCode), null);
+    doThrow(ex).when(mockClient).getAclStatus(anyString(), any(TracingContext.class));
+
+    try {
+      boolean isHnsEnabled = store.getIsNamespaceEnabled(
+          getTestTracingContext(getFileSystem(), false));
+      Assertions.assertThat(isHnsEnabled).isEqualTo(expected);
```

Comment:
Instead of combining two assert cases into one method, can you split into two **ensure*** methods.

Case-1) You can create self-explanatory method saying that, which expects exception.
```
    try {
      boolean isHnsEnabled = store.getIsNamespaceEnabled(
          getTestTracingContext(getFileSystem(), false));
      Assertions.fail("<give proper message>");
    } catch (AbfsRestOperationException caughtEx) {
      Assertions.assertThat(caughtEx.getStatusCode()).isEqualTo(statusCode);
      Assertions.assertThat(caughtEx.getErrorMessage()).isEqualTo(ex.getErrorMessage());
    }
```

case-2)  You can create self-explanatory method saying that, which expects desired boolean flag

```
    try {
      boolean isHnsEnabled = store.getIsNamespaceEnabled(
          getTestTracingContext(getFileSystem(), false));
       Assertions.assertThat(isHnsEnabled).isEqualTo(expected);
    } catch (AbfsRestOperationException caughtEx) {
        Assertions.fail("<give proper message>");
    }
```

Reply:
I have updated the test code to reflect the production code changes.
Let me know if that looks good...